src/osd/OSDMap.cc

```diff
@@ -3081,11 +3014,12 @@ int OSDMap::build_simple_crush_map_from_conf(CephContext *cct,
   // add osds
   vector<string> sections;
   conf->get_all_sections(sections);
-  for (vector<string>::iterator i = sections.begin(); i != sections.end(); ++i) {
-    if (i->find("osd.") != 0)
+
+  for (auto &elem : sections) {
```

Comment:
There are two sections at 3015 and 3029. There is no logical error it seems. But for a reader, it looks confusing.

```  
// add osds
  vector<string> sections;
  conf->get_all_sections(sections);

  for (auto &elem : sections) {
    if (elem.find("osd.") != 0)
      continue;

    const char *begin = elem.c_str() + 4;
    char *end = (char*)begin;
    int o = strtol(begin, &end, 10);
    if (*end != '\0')
      continue;

    string host, rack, row, room, dc, pool;
    vector<string> sections;
    sections.push_back("osd");
sections.push_back(elem);
```